src/QueryBuilder.php

```diff
@@ -28,12 +28,12 @@ final class QueryBuilder extends AbstractQueryBuilder
         SchemaInterface::TYPE_SMALLINT => 'NUMBER(5)',
         SchemaInterface::TYPE_INTEGER => 'NUMBER(10)',
         SchemaInterface::TYPE_BIGINT => 'NUMBER(20)',
-        SchemaInterface::TYPE_FLOAT => 'NUMBER',
-        SchemaInterface::TYPE_DOUBLE => 'NUMBER',
-        SchemaInterface::TYPE_DECIMAL => 'NUMBER',
-        SchemaInterface::TYPE_DATETIME => 'TIMESTAMP',
-        SchemaInterface::TYPE_TIMESTAMP => 'TIMESTAMP',
-        SchemaInterface::TYPE_TIME => 'TIMESTAMP',
+        SchemaInterface::TYPE_FLOAT => 'BINARY_FLOAT',
+        SchemaInterface::TYPE_DOUBLE => 'BINARY_DOUBLE',
+        SchemaInterface::TYPE_DECIMAL => 'NUMBER(10,0)',
+        SchemaInterface::TYPE_DATETIME => 'TIMESTAMP(0)',
+        SchemaInterface::TYPE_TIMESTAMP => 'TIMESTAMP(0)',
+        SchemaInterface::TYPE_TIME => 'INTERVAL DAY(0) TO SECOND(0)',
```

Comment:
I'm not sure. Interval not equivalent of time column

Reply:
There is no `time` type in Oracle and `INTERVAL DAY(0) TO SECOND(0)` looks more appropriate for storing time instead of `TIMESTAMP`

Reply:
+